### ISSUE TITLE
Tabula-java returns "error: null"

### ISSUE THREAD

**Issue by wcraft** [tabulapdf/tabula-java#107]
I have a 1000+ page pdf. I tested tabula-java on a 100 page extract of the whole document, and it worked fine. However, when I ran the entire pdf, the program returned "Error: null" and gave me an empty csv.

**Comment by jazzido** [comment]
Can you send us the PDF, so we can examine the issue for ourselves?

You can send the PDF any of the following three ways:

1. send a link, if the PDF is accessible over the internet
2. attach it via Github's web Issues interface, if the document is not sensitive. (You must use the web interface. If you attempt to attach it to an email reply to GitHub, it will not be sent.)
3. send it directly by email to me (or another Tabula collaborator), using the email address in my profile. If the document is sensitive, please say so.

**Comment by jazzido** [comment]
Hi @wcraft,

I can confirm that `tabula-java` crashes on the following command:

```
java -jar tabula-0.9.1-jar-with-dependencies.jar --spreadsheet -a 99.068,73.823,689.648,526.703 -p all  document.pdf
```

The exception thrown is:

```
Exception in thread "main" java.util.NoSuchElementException
	at java.util.ArrayList$Itr.next(ArrayList.java:854)
	at java.util.Collections.min(Collections.java:635)
	at technology.tabula.Page.getArea(Page.java:67)
	at technology.tabula.CommandLineApp.extractTables(CommandLineApp.java:147)
	at technology.tabula.CommandLineApp.main(CommandLineApp.java:60)
```

The extraction method is crashing on one of the 1191 (!) pages of the document. I'll need to investigate this further to find the cause. Unfortunately, I can't provide a timeline for this fix, but in the meantime, you can try `pdftotext`, included in the Poppler library. The document is quite well structured, so the command `pdftotext -layout document.pdf` seems to produce output that is quite parseable.

Thanks!

**Comment by wcraft** [comment]
Thanks for the help. I will try pdftotext in the meantime

**Comment by jeremybmerrill** [comment]
might be the same as https://github.com/tabulapdf/tabula/issues/573 and https://github.com/tabulapdf/tabula-java/issues/105

**Comment by blomdahldaniel** [comment]
I think that I just got the same issue. 
Any news on this?

**Comment by jeremybmerrill** [comment]
@blomdahldaniel: I think we'll have a fix for this shortly (like, minutes shortly. hopefully). Once #131 is merged in, are you able to build from master yourself? (`mvn` command is in readme)

**Comment by blomdahldaniel** [comment]
Wow! Funny coincidence! Awesome!
No I do not have ability to build from master, or at least I dont know hot to do it. Will you distribute a new version later on that includes this fix? If thats so, could you guess when that could be? I think I might save myself the hustle and wait until you can distribute a version.

While I still got your attention: I just wanna say that tabula is awesome and it helps me so much in my projects! Big thanks!

**Comment by jeremybmerrill** [comment]
Hey, glad it's useful! 

I dunno when exactly we'll push a new version. I can build you a bootleg version if you're okay with downloading a jar from my Dropbox...?

**Comment by blomdahldaniel** [comment]
Would love to, you can email me mail@blomdahldaniel.se

**Comment by jeremybmerrill** [comment]
@blomdahldaniel try this out: https://www.dropbox.com/s/yekob7tvjuw9l1w/tabula-0.9.1-jar-with-dependencies.jar?dl=0

closed via #131

**Comment by blomdahldaniel** [comment]
Very grateful!

**Comment by blomdahldaniel** [comment]
Awesome!
Tabula doesn't break anymore while given an empty area. However now I get the entire page instead..? So, the same area on one pdf that only contains one word, that word is extracted properly. But with another pdf where the same area is blank, I get all existing words in that pdf as a result..?

**Comment by jeremybmerrill** [comment]
@blomdahldaniel Yep, you're right. Can you try this version and let me know if it works? https://www.dropbox.com/s/yekob7tvjuw9l1w/tabula-0.9.1-jar-with-dependencies.jar?dl=0

**Comment by blomdahldaniel** [comment]
Yes this works better I guess, now I receive an exception instead:

```
Error Output:
================
Exception in thread "main" java.lang.IllegalArgumentException: shapes can't be empty
at technology.tabula.Utils.bounds(Utils.java:44)
at technology.tabula.extractors.SpreadsheetExtractionAlgorithm.isTabular(SpreadsheetExtractionAlgorithm.java:151)
at 
```

Is that expected behaviour instead of giving me just a blank/empty result?

In my head, giving me an empty result would make more sense then throwing an exception though..?

**Comment by jeremybmerrill** [comment]
@blomdahldaniel: No, that's not expected behavior. You should get an empty result of some sort. I've only seen that "shapes can't be empty" bug when the page has no text at all. Does that describe one of the pages you're dealing with?

**Comment by jeremybmerrill** [comment]
@blomdahldaniel: Here's another to try, with two more patches: https://www.dropbox.com/s/yekob7tvjuw9l1w/tabula-0.9.1-jar-with-dependencies.jar?dl=0

**Comment by blomdahldaniel** [comment]
To answer your first comment:
The exact same document had text in it that was extracted properly but not when the area/shape was empty. Then I got the exception.

The last patch from dropbox seem to work just as expected in my setup. When I provide an empty area I just receive an empty result, perfect! So as far as I can see this seem to be exactly the wanted result! Good job and big thank you! I can now clean up my code where I handled this issue myself. 😃 🎉
